examples/resnet50_server_pytorch.py

```diff
@@ -32,9 +36,26 @@ class CategoryResp(BaseModel):
 
 
 class Preprocess(Worker):
+    def deserialize(self, data: bytes) -> ImageReq:
+        data_json = super().deserialize(data)  # default json decoder
+
+        # Customized validation for schema, raise ValidationError
+        # so that the client can get 422 as http status
+        try:
+            image_req = ImageReq.parse_obj(data_json)
+        except Exception as err:
+            raise ValidationError(err)
+        return image_req
+
     def forward(self, req: ImageReq) -> np.ndarray:
-        im = np.frombuffer(base64.b64decode(req.image), np.uint8)
-        im = cv2.imdecode(im, cv2.IMREAD_COLOR)[:, :, ::-1]  # bgr -> rgb
+        # Customized validation for content, raise ValidationError
+        # so that the client can get 422 as http status
+        try:
+            im = np.frombuffer(base64.b64decode(req.image), np.uint8)
+            im = cv2.imdecode(im, cv2.IMREAD_COLOR)[:, :, ::-1]  # bgr -> rgb
+        except Exception as err:
+            raise ValidationError(f"cannot decode as image data: {err}")
```

Comment:
`forward` should focus on the logic, not the validation part.

Reply:
Moving it into `deserialize` part? Yea but the content validation may also be considered as part of the logic. Because the user wants to check some specific data requirements, and when they fail, the user wants to raise this error instead of 500.

Reply:
Any other image preprocessing needed here? For example, resizing, cropping, normalization.

Reply:
It's done after this validation.